src/create-model.jl

```diff
@@ -392,22 +448,41 @@ function create_model(
     ## Helper functions
     # Computes the duration of the `block` and multiply by the resolution of the
     # representative period `rp`.
-    function duration(timesteps_block, rp)
+    function duration(timesteps_block, rp, representative_periods)
         return length(timesteps_block) * representative_periods[rp].resolution
     end
+
+    Y = years = collect(keys(representative_periods)) # need to decide where to get the years
+
     # Maximum timestep
-    Tmax = maximum(last(rp.timesteps) for rp in representative_periods)
+    Tmax = maximum(last(rp.timesteps) for year in Y for rp in representative_periods[year])
+
     expression_workspace = Vector{JuMP.AffExpr}(undef, Tmax)
 
     ## Sets unpacking
     @timeit to "unpacking and creating sets" begin
         A = MetaGraphsNext.labels(graph) |> collect
         F = MetaGraphsNext.edge_labels(graph) |> collect
-        filter_assets(key, values) =
-            filter(a -> !ismissing(getfield(graph[a], key)) && getfield(graph[a], key) == values, A)
+        filter_assets(key, value) =
+            filter(a -> !ismissing(getfield(graph[a], key)) && getfield(graph[a], key) == value, A)
         filter_assets(key, values::Vector{String}) =
             filter(a -> !ismissing(getfield(graph[a], key)) && getfield(graph[a], key) in values, A)
-        filter_flows(key, values) = filter(f -> getfield(graph[f...], key) == values, F)
+
+        filter_assets(key, year, value) = filter(
+            a -> begin
+                field_value = get(getfield(graph[a], key), year, missing)
+                get(graph[a].active, year, false) &&
```

Comment:
just a generic comment: if we only use one row for one unit name each year, then `graph[a].active` does not make sense anymore, we need to change this later

Reply:
Agreed. We need to figure out how to deal with the lifetime column and its implications in the model.